### PR TITLE
extraction pipelines

### PR DESCRIPTION
**Pull request by KissSpinobleZsolt**

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -139,6 +118,23 @@ export const defaultTemplateQuery: TemplateQuery = {
...
+  getRuns: false,
+  columns: [
+    'name',
+    'status',
+    'lastUpdatedTime',
+    'lastFailure',
+    'lastSeen',
+    'lastSuccess',
+    'schedule',
+    'data Set',
```
Comment (polomani) [inline src/types.ts:132]:
```suggestion
    'data set',
```

File: src/constants.ts [hunk 2]
```diff
@@ -30,21 +30,56 @@ export const AuthType = {
...
+const createFields = (parent, field) => `${parent}-${field}`;
+const metaFields = [
+  'Dockerfile',
+  'Extractor config',
+  'branch',
+  'environment',
+  'extractor',
+  'site',
+  'source',
+].map((_) => createFields('metadata', _));
```
Comment (polomani) [inline src/constants.ts:63]: are those some customer specific fields, or is a part of common practice for the extraction pipelines?

File: src/__tests__/datasource.test.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@ type QueryTargetLike = Partial<CogniteQuery>;
 
 const ds = getMockedDataSource();
 const { backendSrv, templateSrv } = ds;
-const { Asset, Custom, Timeseries } = Tab;
+const { Asset, Custom, Timeseries, ExtractionPipelines } = Tab;
```
Comment (polomani) [inline src/__tests__/datasource.test.ts:17]:
```suggestion
const { Asset, Custom, Timeseries } = Tab;
```
doesn't seem to be used?

File: src/datasources/ExtractionPipelinesDatasource.ts [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+          )
+        );
+      } else if (_.isObject(value)) {
+        _.map(value, (v, k) => _.assignIn(resource, { [`${key}-${k}`]: v }));
+      } else _.assignIn(resource, { [key]: value });
+    });
+
+    return resource as Resource;
+  });
+  const tabelData = convertItemsToTable(
```
Comment (polomani) [inline src/datasources/ExtractionPipelinesDatasource.ts:50]:
```suggestion
  const tableData = convertItemsToTable(
```

File: src/datasources/ExtractionPipelinesDatasource.ts [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+              1
+            );
+            data = {
+              externalId,
+              ...rest,
+              status: response[0]?.status,
+              message: response[0]?.message,
+              runId: response[0]?.id,
+            };
+            if (columns.includes('data Set')) {
```
Comment (polomani) [inline src/datasources/ExtractionPipelinesDatasource.ts:144]:
```suggestion
            if (columns.includes('data set')) {
```

**Review by polomani — COMMENTED** [review]
I know extraction pipelines are in GA, but considering the amount of tabs we have now and also the fact that this specific API has a specific use-case and targeted to a specific subset of users of the connector I would sugget making this as an opti-in feature.

(same as we did for templates/flexible models)

File: src/constants.ts (hunk 2 above)
Comment (KissSpinobleZsolt) [reply src/constants.ts:63]: they are the object as value from the Resource response which will be supported under the table as values by 'sourceKey-key' , this is the declaration of tableColumName for the key metadata from the EP response Object, to be present when adding new column to table.

File: src/components/configEditor.tsx [hunk 6]
```diff
@@ -36,6 +36,8 @@ const enableEventsAdvancedFilteringTooltip = `Enable the Events advanced filteri
 
 const enableFlexibleDataModellingTooltip = 'Enable Flexible Data Modelling (preview)';
 
+const enableExtractionPipalinesTooltip = 'Enable Extraction Pipalines (preview)';
```
Comment (polomani) [inline src/components/configEditor.tsx:39]:
```suggestion
const enableExtractionPipalinesTooltip = 'Enable Extraction Pipelines (preview)';
```

File: src/components/configEditor.tsx [hunk 7]
```diff
@@ -54,6 +56,7 @@ export function ConfigEditor(props: ConfigEditorProps) {
     enableTemplates,
     enableEventsAdvancedFiltering,
     enableFlexibleDataModelling,
+    enableExtractionPipalines,
```
Comment (polomani) [inline src/components/configEditor.tsx:59]:
```suggestion
    enableExtractionPipelines,
```

File: src/components/configEditor.tsx [hunk 8]
```diff
@@ -255,6 +258,15 @@ export function ConfigEditor(props: ConfigEditorProps) {
             tooltip={enableFlexibleDataModellingTooltip}
           />
         </div>
+        <div className="gf-form-inline">
+          <Switch
+            label="Extraction Pipalines"
+            labelClass="width-11"
+            checked={enableExtractionPipalines}
+            onChange={onJsonBoolValueChange('enableExtractionPipalines')}
```
Comment (polomani) [inline src/components/configEditor.tsx:266]:
```suggestion
            onChange={onJsonBoolValueChange('enableExtractionPipelines')}
```

File: src/types.ts [hunk 9]
```diff
@@ -171,6 +168,7 @@ export interface CogniteDataSourceOptions extends DataSourceJsonData {
   enableTemplates?: boolean;
   enableEventsAdvancedFiltering?: boolean;
   enableFlexibleDataModelling?: boolean;
+  enableExtractionPipalines?: boolean;
```
Comment (polomani) [inline src/types.ts:171]:
```suggestion
  enableExtractionPipelines?: boolean;
```

**Review by polomani — APPROVED** [review]
LGTM

**Pull request merged**